### PR TITLE
Feature/drafts

### PR DESCRIPTION
**Pull request by DJankauskas**
Filter out unpublished articles from being shown to website visitors, unless they have a direct link to an unpublished article. Also allows specifying articles as unpublished or published, which will allow for creating unpublished articles throughout the issue lifecycle, then publishing them in one batch.

### CODE REVIEW

File: app/graphql/resolvers/article_query_function.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Resolvers::ArticleQueryFunction < GraphQL::Function
+
+    def select_published(articles)
+        # nil articles will be filtered out
+        if articles.nil?
```
Comment (NicholasLYang) [inline app/graphql/resolvers/article_query_function.rb]: `return nil if articles.nil?` is more ruby-esque

File: app/graphql/resolvers/article_query_function.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Resolvers::ArticleQueryFunction < GraphQL::Function
+
+    def select_published(articles)
+        # nil articles will be filtered out
+        if articles.nil?
+            return nil
+        end
+        
+        return articles.where(is_published: true)
```
Comment (NicholasLYang) [inline app/graphql/resolvers/article_query_function.rb]: imo this should be in a model

File: app/graphql/resolvers/article_query_function.rb (hunk 1 above)
Comment (DJankauskas) [reply app/graphql/resolvers/article_query_function.rb]: Thanks for the suggestion, I changed it.

File: app/graphql/resolvers/article_query_function.rb (hunk 2 above)
Comment (DJankauskas) [reply app/graphql/resolvers/article_query_function.rb]: Could you explain exactly what you mean? I'm not familiar with models. Thanks!

Comment (NicholasLYang) [reply app/graphql/resolvers/article_query_function.rb]: In Rails there's this idea that you should have thin controllers and fat models. Models contain the logic for the actual piece of data, while controllers contain the logic for connecting the models to the views. In general, controllers/resolvers shouldn't have a whole lot of logic in them. Instead, it should be encapsulated in the model. The model is in `app/models/article.rb`. In this case, if you want to get the published articles, maybe you want to make a class method like:

```
class Article

def published()
  Article.where(is_published: true)
end

end
```
Then you can do:
```
Article.published.first(10)
```

Or whatever

File: app/graphql/resolvers/create_article.rb [hunk 3]
```diff
@@ -31,6 +32,7 @@ def call(_obj, args, ctx)
         issue: args["issue"],
         summary: args["summary"],
         created_at: args["created_at"],
+        is_published: args["is_published"] || true
```
Comment (NicholasLYang) [inline app/graphql/resolvers/create_article.rb]: Do you really want to default to true?

File: app/graphql/resolvers/get_featured_articles_by_section_id.rb [hunk 4]
```diff
@@ -10,23 +10,27 @@ class Resolvers::GetFeaturedArticlesBySectionID < GraphQL::Function
...
       Article
         .joins('JOIN sections ON articles.section_id = sections.id')
         .where("sections.id = #{args['section_id']} AND articles.id != #{primary_article.id}")
         .order("articles.rank + 3 * sections.rank + 12 * articles.issue"\
                " + 192 * articles.volume DESC")
+    )
 
     if primary_article.nil?
-      return secondary_articles.first(3)
+      return select_published(secondary_articles).first(3)
```
Comment (NicholasLYang) [inline app/graphql/resolvers/get_featured_articles_by_section_id.rb]: Seems like you're calling `select_published` twice

File: app/graphql/resolvers/article_query_function.rb (hunk 2 above)
Comment (DJankauskas) [reply app/graphql/resolvers/article_query_function.rb]: That's a lot nicer, I changed all my code to use that, and added an unpublished method as well.

File: app/graphql/resolvers/create_article.rb (hunk 3 above)
Comment (DJankauskas) [reply app/graphql/resolvers/create_article.rb]: Eventually I'll remove or change the default, but for now I want code currently in use to work without modifications.

File: app/graphql/resolvers/get_featured_articles_by_section_id.rb (hunk 4 above)
Comment (DJankauskas) [reply app/graphql/resolvers/get_featured_articles_by_section_id.rb]: Yeah, that should be fixed.

**Pull request merged**